src/script/benches/py_benchmark.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use catalog::local::{MemoryCatalogProvider, MemorySchemaProvider};
+use catalog::{CatalogList, CatalogProvider, SchemaProvider};
+use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
+use common_query::Output;
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use futures::Future;
+use once_cell::sync::{Lazy, OnceCell};
+use query::QueryEngineFactory;
+use rayon::ThreadPool;
+use script::engine::{CompileContext, EvalContext, Script, ScriptEngine};
+use script::python::PyEngine;
+use table::table::numbers::NumbersTable;
+use tokio::runtime::Runtime;
+
+static SCRIPT_ENGINE: Lazy<PyEngine> = Lazy::new(sample_script_engine);
+static LOCAL_RUNTIME: OnceCell<tokio::runtime::Runtime> = OnceCell::new();
+fn get_local_runtime() -> std::thread::Result<&'static Runtime> {
+    let rt = LOCAL_RUNTIME
+        .get_or_try_init(|| tokio::runtime::Runtime::new().map_err(|e| Box::new(e) as _))?;
+    Ok(rt)
+}
+/// a terrible hack to call async from sync by:
+/// TODO(discord9): find a better way
+/// 1. spawn a new thread
+/// 2. create a new runtime in new thread and call `block_on` on it
+pub fn block_on_async<T, F>(f: F) -> std::thread::Result<T>
+where
+    F: Future<Output = T> + Send + 'static,
+    T: Send + 'static,
+{
+    let rt = get_local_runtime()?;
+
+    std::thread::spawn(move || rt.block_on(f)).join()
+}
+
+pub(crate) fn sample_script_engine() -> PyEngine {
+    let catalog_list = catalog::local::new_memory_catalog_list().unwrap();
+
+    let default_schema = Arc::new(MemorySchemaProvider::new());
+    default_schema
+        .register_table("numbers".to_string(), Arc::new(NumbersTable::default()))
+        .unwrap();
+    let default_catalog = Arc::new(MemoryCatalogProvider::new());
+    default_catalog
+        .register_schema(DEFAULT_SCHEMA_NAME.to_string(), default_schema)
+        .unwrap();
+    catalog_list
+        .register_catalog(DEFAULT_CATALOG_NAME.to_string(), default_catalog)
+        .unwrap();
+
+    let factory = QueryEngineFactory::new(catalog_list);
+    let query_engine = factory.query_engine();
+
+    PyEngine::new(query_engine.clone())
+}
+async fn run_script(script: &str) {
+    let script = script;
+    let script = SCRIPT_ENGINE
```

Comment:
In fact, we cache the compiled result in memory, and execute it many times, can we simulate this behaviour? Cache the compiled result too.

Reply:
> In fact, we cache the compiled result in memory, and execute it many times, can we simulate this behaviour? Cache the compiled result too.

Did that, choosed compile once and run for ten times repeatly